snapshot/backup.go

```diff
@@ -435,18 +444,33 @@ func (snap *Builder) chunkify(imp importer.Importer, record *importer.ScanRecord
 				break
 			}
 
-			chunkCopy := make([]byte, len(cdcChunk))
-			copy(chunkCopy, cdcChunk)
+			objectHasher.Write(cdcChunk)
 
-			objectHasher.Write(chunkCopy)
+			chunk := objects.NewChunk()
+			chunk.Length = uint32(len(cdcChunk))
+			object.Chunks = append(object.Chunks, *chunk)
+
+			cdcCopy := make([]byte, len(cdcChunk))
+			copy(cdcCopy, cdcChunk)
+
+			semaphore <- struct{}{}
+			wg.Add(1)
+
+			go func(idx int, cdcCopy []byte) {
+				defer wg.Done()
+				defer func() { <-semaphore }()
+
+				if err := processChunk(idx, cdcCopy); err != nil {
+					//return nil, err
+				}
```

Comment:
not a huge fan of this to be honest :D

we need to have a way to handle the PutBlob failure.

Reply:
I was just making sure people reviewed my diffs :-p

Reply:
fixed by switching to an errgroup